fobi_custom/plugins/form_handlers/collect_data/fobi_form_handlers.py

```diff
@@ -69,53 +69,51 @@ def run(self, form_entry, request, form, form_element_entries=None):
                     saved_data=element_info['saved_data']
                 )
 
+    def get_element_info(self, form_element):
+        plugin_data = form_element.plugin_data
+        json_plugin_data = json.loads(plugin_data)
 
-def plugin_data_repr(self):
-    """Human readable representation of plugin data.
-
-    :return string:
-    """
-    return self.data.__dict__
-
+        if 'text' in json_plugin_data.keys():
+            return None
 
-def get_element_info(self, form_element):
-    plugin_data = form_element.plugin_data
-    json_plugin_data = json.loads(plugin_data)
+        else:
+            name = json_plugin_data['name']
 
-    if 'text' in json_plugin_data.keys():
-        return None
+            element_info = {
+                'name': name,
+                'saved_data': self.form.cleaned_data[name],
+                'label': json_plugin_data['label'],
+                'type': form_element.plugin_uid,
+                'position': form_element.position,
+            }
 
-    else:
-        name = json_plugin_data['name']
+            return element_info
 
-        element_info = {
-            'name': name,
-            'saved_data': self.form.cleaned_data[name],
-            'label': json_plugin_data['label'],
-            'type': form_element.plugin_uid,
-            'position': form_element.position,
-        }
+    def get_element_data(self, form_element):
+        element_info = self.get_element_info(form_element)
+        saved_data = element_info['saved_data']
 
-        return element_info
+        return saved_data
 
+    def get_saved_data(self, plugin_uid, default_value=''):
+        try:
+            form_element = FormElementEntry.objects.get(form_entry_id=self.form_id, plugin_uid=plugin_uid)
+        except FormElementEntry.DoesNotExist:
+            saved_data = default_value
+        else:
+            saved_data = self.get_element_data(form_element)
+            if (plugin_uid in ['time_start', 'time_stop']) and saved_data:
+                saved_data = datetime.combine(self.today, saved_data).replace(tzinfo=self.timezone)
 
-def get_element_data(self, form_element):
-    element_info = get_element_info(self, form_element)
-    saved_data = element_info['saved_data']
+        return saved_data
 
-    return saved_data
 
+def plugin_data_repr(self):
```

Comment:
I would expect this to be a method on the class too, but it looks like it's dedented. Is there a reason for that?

Reply:
This is a holdover from Fobi! I don't see a need to mess with it, since it works and I'm not sure where else it's used in the package.

Reply:
Sounds great, thanks for clarifying.